src/wp-includes/l10n.php

```diff
@@ -893,6 +903,9 @@ function load_plugin_textdomain( $domain, $deprecated = false, $plugin_rel_path
 		$path = WP_PLUGIN_DIR;
 	}
 
+	/* @var WP_Textdomain_Registry $wp_textdomain_registry */
+	$wp_textdomain_registry->set( $domain, $path );
```

Comment:
Do we need this if it's already called in `load_textdomain()`?

Reply:
`load_textdomain` calls it quite late, and potentially not at all if the file does not exist.

Looking at this, the use case this aims to cove is this:

1. Plugin ships with de_DE translation files. Current locale is es_ES.
1. Plugin calls `load_plugin_textdomain()` - no translations are loaded
1. Locale is switched to de_DE - de_DE translation file should be loaded

Without this call here, we'd never know where to look  for the de_DE file because the path wasn't in the registry.